includes/Integrations/Core_Themes_Support.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Class Core_Themes_Support
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Integrations;
+
+use Google\Web_Stories\Customizer;
+
+/**
+ * Class Core_Themes_Support.
+ */
+class Core_Themes_Support {
+
+	/**
+	 * Current theme
+	 *
+	 * @var string
+	 */
+	protected $template = '';
+
+	/**
+	 * Default array of core themes to add support to.
+	 *
+	 * @var array
+	 */
+	protected static $supported_themes = [
+		'twentytwentyone',
+		'twentytwenty',
+		'twentynineteen',
+		'twentyseventeen',
+		'twentysixteen',
+		'twentyfifteen',
+		'twentyfourteen',
+		'twentythirteen',
+		'twentytwelve',
+		'twentyeleven',
+		'twentyten',
+		'classic',
+		'default',
+	];
+
+	/**
+	 * Get list of supported core themes.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return string[] Slugs for supported themes.
+	 */
+	public static function get_supported_themes() {
+		return self::$supported_themes;
+	}
+
+	/**
+	 * Adds theme support for Web Stories.
+	 *
+	 * This will enable add_theme_support with predefined
+	 * options supported themes.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return void
+	 */
+	public function extend_theme_support() {
+		add_theme_support( 'web-stories' );
+	}
+
+	/**
+	 * Embed Webstories.
+	 *
+	 * Embeds web stories with default customizer settings.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return void
+	 */
+	public function embed_web_stories() {
+
+		if ( class_exists( 'Google\Web_Stories\Customizer' ) ) {
+			$customizer = new Customizer();
+			echo $customizer->render_stories(); // phpcs:ignore -- WordPress.Security.EscapeOutput.OutputNotEscaped - Escaped web stories HTML.
+		}
+
+	}
+
+	/**
+	 * Sets current theme.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return string
+	 */
+	public function get_current_theme() {
+		return get_template();
+	}
+
+	/**
+	 * Adds theme support and hook to embed the web stories.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @return void
+	 */
+	public function init() {
+
+		$this->template = $this->get_current_theme();
+
+		if ( ! in_array( $this->template, self::get_supported_themes(), true ) ) {
```

Comment:
```suggestion
		if ( ! in_array( get_template(), self::get_supported_themes(), true ) ) {
```

Wouldn't this make more sense?

Reply:
Confirm that `get_template` get parent theme if the current theme is a child theme. Is this the expected behaviour? 

> In the case a child theme is being used, the directory name of the parent theme will be returned. 

https://developer.wordpress.org/reference/functions/get_template/

Reply:
will replace this with `get_stylesheet()`.

Reply:
@swissspidy What you think about supporting child themes?

Reply:
> So to identify that it's a core theme and is a main template we used `get_template`

@divyarajmasani wouldn't you need to use `get_stylesheet()` for that?

If I am not mistaken:

If I use a **Twenty Ten Child Theme**:

* `get_template()` returns `twentyten`
* `get_stylesheet()` returns `twentyten-child-theme`

If I use **Twenty Ten**:

* `get_template()` returns `twentyten`
* `get_stylesheet()` returns `twentyten`

So this current code (`get_template()`) means it will also work for child themes.

There is no guarantee it's going to work for child themes as well, so I'd avoid supporting that for now. Which means we would need to use `get_stylesheet()` here.

Reply:
You are right, I had thought of extending the support for the child themes as well, but as you mentioned, we won't know the particulars of the child theme. I'll update this to use `get_stylesheet()`